lib/spack/spack/cmd/create.py

```diff
@@ -818,10 +909,56 @@ def get_versions(args, name):
     # FIXME: Add proper versions here.
     # version('1.2.4')"""
 
+    # Default git-based version
+    git_versions = """\
+    # FIXME: Add proper versions referencing branch/tag/commit here
+    # version('main', branch='main)"""
+
     # Default guesser
     guesser = BuildSystemGuesser()
 
     valid_url = True
+
+    url, git = get_url_and_git(args)
+
+    has_git_option = args.commit is not None or \
+        args.tag is not None or \
+        args.branch is not None
```

Comment:
This could be tightened:

```suggestion
    has_git_option = args.commit or args.tag or args.branch
```